From 76c5fadce474b721e429fd4f10ada80ac702037b Mon Sep 17 00:00:00 2001
From: ztong <Zhe.Tong@synaptics.com>
Date: Wed, 14 Dec 2022 14:53:08 +0800
Subject: [PATCH 4/7] avcodec/ac3: fix the larger BD issue

- Add header parser for big/little endian checking in ac3 parser
Original CL: https://sc-debu-git.synaptics.com/gerrit/59062

Change-Id: I61a9a9425fab6efa23052e441380319e45c6e8f9
Reviewed-on: https://gerrit-sha.synaptics.com/gerrit/176775
Reviewed-by: Zhe Tong <Zhe.Tong@synaptics.com>
Reviewed-by: Yanyan Peng <Yanyan.Peng@synaptics.com>
---
 libavcodec/ac3_parser.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavcodec/ac3_parser.c b/libavcodec/ac3_parser.c
index 4f154bb7c4..cc898bf106 100644
--- a/libavcodec/ac3_parser.c
+++ b/libavcodec/ac3_parser.c
@@ -61,7 +61,7 @@ int ff_ac3_parse_header(GetBitContext *gbc, AC3HeaderInfo *hdr)
     memset(hdr, 0, sizeof(*hdr));
 
     hdr->sync_word = get_bits(gbc, 16);
-    if(hdr->sync_word != 0x0B77)
+    if((hdr->sync_word != 0x0B77) && (hdr->sync_word != 0x770B))
         return AAC_AC3_PARSE_ERROR_SYNC;
 
     /* read ahead to bsid to distinguish between AC-3 and E-AC-3 */
-- 
2.17.1

